src/main/java/teammates/ui/webapi/output/StudentData.java

```diff
@@ -116,12 +116,29 @@ public void hideLastName() {
         setLastName(null);
     }
 
+    /**
+     * Hides some attributes to instructor on a search.
+     */
+    public void hideInformationForInstructorSearch() {
```

Comment:
This is not necessary if the information is not sensitive because:

- It does not save any DB retrieval cost
- It does save the byte transferred, but comments and institute are never that large to warrant that kind of saving
- These information are accessible from some other instructor endpoints that are not search, which defeats any goal of information hiding
- Having this method requires some maintenance effort, e.g. "why these fields?", "why only for instructor search?"

Reply:
There was already a method called `hideInformationForInstructor` which wasn't appropriately named because it was only used in the student search so I just renamed it to `hideInformationForInstructorSearch`. 

I removed `googleId` from being set in the constructor and renamed the methods more appropriately. I'll fix the tests first then push the changes.